pkg/network/setup/filters.go

```diff
@@ -0,0 +1,34 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors
+ *
+ */
+
+package network
+
+import (
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
+)
+
+func FilterNetsForVMStartup(vmi *v1.VirtualMachineInstance) []v1.Network {
```

Comment:
How about `FilterNonAbsentNetworks`, `FilterUnpluggedNetworks`?

Asking to understand: could you please remind me what will happen if a VirtualMachine manifest has absent networks, on startup?
Wouldn't the VM controller create a VirtualMachineInstance spec w/o the absent networks?

Reply:
> How about FilterNonAbsentNetworks, FilterUnpluggedNetworks?
The names of the functions imply the stage in the VM life cycle on which they should be called, and not what they do.

Reply:
> Asking to understand: could you please remind me what will happen if a VirtualMachine manifest has absent networks, on startup?
> Wouldn't the VM controller create a VirtualMachineInstance spec w/o the absent networks?

I need to check, but in the context of this PR it doesn't matter - because code is just extracted without logic changed.

Reply:
Please renaming this function so it doesn't describe the context it should be called (i.e: "ForVMStartup").
The context should be realized from the caller. 
Having the name not mentioning the context makes this function more portable.

One correction to the above naming suggestions: `FilterNonAbsentNetworks`, `FilterOutUnpluggedNetworks` or `FilterOutRemovedNetworks`

Reply:
In the VM controller, the network setup is called in three places:
1. When the VM is started for the first time.
2. When the VM is already running - for hotplug / hotunplug of non-SR-IOV NICs.
3. When the VM in started as a migration target.

These filters are only relevant for this single use and are not meant to be reusable.

A follow-up PR is in the works to abstract away these implementation details and give the VM controller three entry-points for the network setup.

Reply:
I dont know what you plan and whats going to be abstracted, I am reviewing this PR and I think the new functions names should avoid saying where they are/should be called (It can be understood by their caller).

Reply:
The functions were put in the `pkg/network/setup` package, which is responsible for the non-SR-IOV network setup.

I don't think one would want to use them in a context other than network setup.

Reply:
@ormergi , when you read the function name, do you understand what it is suppose to do?
In case the author intention was to use it only for a specific case and document it in the name this, what other alternative is there?

Bottom line: Is this function aimed to be specifically used for VM starup or not?
If the abstraction level is at the same level, I do not see a problem here.

You could argue that another abstraction level can be added, so one will wrap the other, but it does not seem useful here (e.g. `FilterNetsForVMStartup` calling `FilterNonAbsentNetworks`).

Reply:
I know what it does because I am familiar with the code. You cant really understand what will be the product from the name (that is non-absent networks).

The caller is one huge function, there is no proper abstraction that let you distinct between VM  startup or update flows easily (it depends on the if clause you point at..).
Having the function say where it should be called feels like enforcing it because the caller dont provider proper abstraction. All I am asking is to avoid that.

Assuming we had such additional abstraction at the caller, the "FromVMStartap" part would be redundant.
FWIW this is what I meant earlier saying "making the function portable", if tomorrow the caller changes no need to rename this function to fit the new changes.